File: TwoSum.py
"""
Given an array of integers nums and an integer target, return 
indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, 
and you may not use the same element twice.

You can return the answer in any order.
"""


def twoSum(self, nums: List[int], target: int) -> List[int]:

  left_pointer = 0
  right_pointer = left_pointer + 1

  while left_pointer < right_pointer:
    total = nums[left_pointer] + nums[right_pointer]
    if total == target:
      return [left_pointer, right_pointer]
    else:
      if(right_pointer == len(nums)-1):
        left_pointer+=1
        right_pointer = left_pointer+1
      else:
        right_pointer+=1
        

File: TwoSum.js
/* 
Given an array of integers nums and an integer target, return 
indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, 
and you may not use the same element twice.

You can return the answer in any order.

*/

const twoSum = function (nums, target) {
  leftPointer = 0;
  rightPointer = leftPointer + 1;

  while (leftPointer < rightPointer) {
    let sum = nums[leftPointer] + nums[rightPointer];
    if (sum === target) return [leftPointer, rightPointer];
    else {
      if (rightPointer == nums.length - 1) {
        leftPointer++;
        rightPointer = leftPointer + 1;
      } else {
        rightPointer++;
      }
    }
  }
};
